dockstore-client/src/main/java/io/dockstore/client/cli/nested/AbstractEntryClient.java

```diff
@@ -520,6 +521,184 @@ private void handleEntry2tsv(List<String> args) throws ApiException, IOException
+                if(matchVersion.find() && !versionFound){
+                    versionFound = true;
+                } else {
+                    if(getEntryType().toLowerCase().equals("workflow") && matchWf.find() && versionFound){
+                        return true;
+                    } else if(getEntryType().toLowerCase().equals("tool") && matchTool.find() && versionFound){
+                        return true;
+                    }
+                    errorMessage("Entry type does not match the class specified in CWL file.",CLIENT_ERROR);
+                }
+            }
+            bufferedReader.close();
+        } catch (IOException e){
+            //e.printStackTrace();
+        }
+        return false;
+    }
+
+    /**
+     * this function will check if the content of the file is WDL or not
+     * @param content
+     * @return
+     */
+    private Boolean checkWDL(File content){
+        /* WDL: check for 'task' (can be >=1) and 'workflow', cause on the description it says that you can construct workflow from tasks
+                and wdl describes tasks and workflows */
+        Pattern taskPattern = Pattern.compile("(.*)(task)(\\s)(.*)(\\{)");
+        Pattern wfPattern = Pattern.compile("(.*)(workflow)(\\s)(.*)(\\{)");
+        String line;
+        Integer counter = 0;
+        try{
+            InputStreamReader is = new InputStreamReader(new FileInputStream(content.getPath()),StandardCharsets.UTF_8);
+            BufferedReader bufferedReader = new BufferedReader(is);
+            ArrayList<String> lines = new ArrayList<>();
+            while((line = bufferedReader.readLine()) != null) {
+                Matcher m = taskPattern.matcher(line);
+                lines.add(line);
+                if (m.find()) {
+                    counter++;
+                }
+            }
+            bufferedReader.close();
+            //check if 'workflow' exist in the file or not
+            if(counter>0){
+                for(String l : lines) {
+                    Matcher m = wfPattern.matcher(l);
+                    if(m.find()){
+                        return true;
+                    }
+                }
+                return false; // 'workflow' does not exist, file is invalid
+            }
+        } catch (IOException e){
+            //e.printStackTrace();
+        }
+        return false;
+    }
+
+
+    /**
+     * this function will check the content of the entry file if it's a valid cwl/wdl file
+     * TO DO : get the pattern of a cwl/wdl file to determine the file content
+     */
+    public String checkFileContent(File content){
+        if(checkWDL(content)){
+            return "wdl";
+        }else if(checkCWL(content)){
+            return "cwl";
+        }
+        return null;
+    }
+
+    /**
+     * this function will check the extension of the entry file (cwl/wdl)
+     * @return String
+     */
+    public String checkFileExtension(String filename){
```

Comment:
Give this a shot https://commons.apache.org/proper/commons-io/javadocs/api-2.5/org/apache/commons/io/FilenameUtils.html#getExtension%28java.lang.String%29

Also account for upper/lowercase